### PR TITLE
add synthetics multi factor authentication docs

### LINKED ISSUE #4334: [Request]: Document MFA TOTP based Auth for Synthetics browser monitors
### Description

Users can now [automatically generate](https://github.com/elastic/synthetics/pull/957) and utilize time-based one-time passwords (TOTP) as a form of multi-factor authentication (MFA) in Synthetics browser journeys. This feature is available for both inline and project-based browser monitors, allowing users to fully test websites protected by MFA without needing to disable security features or manually generate tokens using external tools

### Resources

https://github.com/elastic/synthetics/pull/957

### Which documentation set does this change impact?

Stateful and Serverless

### Feature differences

Identical. 

### What release is this request related to?

8.16

### Collaboration model

The engineering team

### Point of contact.

**Main contact:** @vigneshshanmugam 

**Stakeholders:** @vigneshshanmugam @bturquet @drewpost

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
## Description

Add docs around the TOTP based MFA support in Synthetics https://github.com/elastic/synthetics/pull/957 and https://github.com/elastic/synthetics/pull/975 

### Documentation sets edited in this PR

- [x] Stateful (`docs/en/observability/*`)
- [x] Serverless (`docs/en/serverless/*`)
- [ ] Integrations Developer Guide (`docs/en/integrations/*`)
- [ ] None of the above

### Related issue
Closes #4334 

## Checklist

- [ ] Product/Engineering Review
- [ ] Writer Review

### Follow-up tasks

* This PR does _not_ need to be ported to another doc set because:
  - [ ] The concepts in this PR only apply to one doc set (serverless _or_ stateful)
  - [x] The PR contains edits to both doc sets (serverless _and_ stateful)
* This PR needs to be ported to another doc set:
  - [ ] Port to stateful docs: \<link to PR or tracking issue>
  - [ ] Port to serverless docs: \<link to PR or tracking issue>

### CODE REVIEW

File: docs/en/observability/synthetics-mfa.asciidoc [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+```sh
+npx @elastic/synthetics totp <secret>
+
+// prints
+OTP Token: 123456
+```
+
+== Using TOTP token inside the journeys
+
+Once the Synhetics TOTP Authentication is configured in your application, you
```
Comment (drewpost) [inline docs/en/observability/synthetics-mfa.asciidoc:32]: typo here/ Should be "Synthetics"

File: docs/en/observability/synthetics-mfa.asciidoc [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+= Multi-factor Authentication (MFA) for browser monitors
+
+++++
+<titleabbrev>Multi-factor Authentication</titleabbrev>
+++++
+
+Multi-factor Authentication (MFA) adds an essential layer of security to
+applications login processes, protecting against unauthorized access. A very
+common usecase in Synthetics is testing user journeys that involves websites
+behind protected behind MFA.
```
Comment (emilioalvap) [inline docs/en/observability/synthetics-mfa.asciidoc:11]:
```suggestion
common usecase in Synthetics is testing user journeys that involves websites
protected behind MFA.
```

File: docs/en/serverless/synthetics/synthetics-mfa.mdx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+<p>
+  <DocBadge template="technical preview" />
+</p>
+
+<div id="synthetics-mfa"></div>
+
+Multi-factor Authentication (MFA) adds an essential layer of security to
+applications login processes, protecting against unauthorized access. A very
+common usecase in Synthetics is testing user journeys that involves websites
+behind protected behind MFA.
```
Comment (emilioalvap) [inline docs/en/serverless/synthetics/synthetics-mfa.mdx:17]:
```suggestion
common usecase in Synthetics is testing user journeys that involves websites
protected behind MFA.
```

File: docs/en/observability/synthetics-mfa.asciidoc (hunk 1 above)
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-mfa.asciidoc:32]:
```suggestion
Once the Synthetics TOTP Authentication is configured in your application, you
```

File: docs/en/serverless/synthetics/synthetics-mfa.mdx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+```sh
+npx @elastic/synthetics totp <secret>
+
+// prints
+OTP Token: 123456
+```
+
+## Using TOTP token inside the journeys
+
+Once the Synhetics TOTP Authentication is configured in your application, you can now use the OTP token in the synthetics browser
```
Comment (vigneshshanmugam) [inline docs/en/serverless/synthetics/synthetics-mfa.mdx:38]:
```suggestion
Once the Synthetics TOTP Authentication is configured in your application, you can now use the OTP token in the synthetics browser
```

File: docs/en/observability/synthetics-mfa.asciidoc [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+Multi-factor Authentication (MFA) adds an essential layer of security to
+applications login processes, protecting against unauthorized access. A very
+common usecase in Synthetics is testing user journeys that involves websites
+protected behind MFA.
+
+Synthetics supports testing websites secured by Time-based One-Time Password
+(TOTP), a common MFA method that provides short lived One Time tokens that are
+desirable for enhancing security.
+
+== Configuring using TOTP
```
Comment (colleenmcginnis) [inline docs/en/observability/synthetics-mfa.asciidoc:17]:
```suggestion
[discrete]
== Configuring using TOTP
```

File: docs/en/observability/synthetics-mfa.asciidoc [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+Time Password (OTP) using the Synthetics CLI, refer to <<elastic-synthetics-totp-command>>.
+
+```sh
+npx @elastic/synthetics totp <secret>
+
+// prints
+OTP Token: 123456
+```
+
+== Using TOTP token inside the journeys
```
Comment (colleenmcginnis) [inline docs/en/observability/synthetics-mfa.asciidoc:30]:
```suggestion
[discrete]
== Using TOTP token inside the journeys
```

**Review by colleenmcginnis — COMMENTED** [review]
Currently each section is on its own page ([preview](https://observability-docs_bk_4444.docs-preview.app.elstc.co/guide/en/observability/master/synthetics-mfa.html)). I think they can go on one page (using the `discrete` tags below). What do you think?

File: docs/en/observability/synthetics-mfa.asciidoc [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+++++
+
+Multi-factor Authentication (MFA) adds an essential layer of security to
+applications login processes, protecting against unauthorized access. A very
+common usecase in Synthetics is testing user journeys that involves websites
+protected behind MFA.
+
+Synthetics supports testing websites secured by Time-based One-Time Password
+(TOTP), a common MFA method that provides short lived One Time tokens that are
+desirable for enhancing security.
```
Comment (bmorelli25) [inline docs/en/observability/synthetics-mfa.asciidoc:15]: I think we can simplify this to:

```suggestion
(TOTP), a common MFA method that provides short-lived one-time tokens to
enhance security.
```

File: docs/en/serverless/synthetics/synthetics-mfa.mdx [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+<div id="synthetics-mfa"></div>
+
+Multi-factor Authentication (MFA) adds an essential layer of security to
+applications login processes, protecting against unauthorized access. A very
+common usecase in Synthetics is testing user journeys that involves websites
+protected behind MFA.
+
+Synthetics supports testing websites secured by Time-based One-Time Password
+(TOTP), a common MFA method that provides short lived One Time tokens that are
+desirable for enhancing security.
```
Comment (bmorelli25) [inline docs/en/serverless/synthetics/synthetics-mfa.mdx:21]:
```suggestion
(TOTP), a common MFA method that provides short-lived one-time tokens to
enhance security.
```

File: docs/en/observability/synthetics-mfa.asciidoc [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+== Configuring using TOTP
+
+To test a browser journey that uses TOTP for MFA, you need to first configure the
+Synthetics authenticator token in the target application. To do that, you can generate a One
+Time Password (OTP) using the Synthetics CLI, refer to <<elastic-synthetics-totp-command>>.
```
Comment (bmorelli25) [inline docs/en/observability/synthetics-mfa.asciidoc:21]:
```suggestion
To test a browser journey that uses TOTP for MFA, first configure the
Synthetics authenticator token in the target application. To do this, generate a One-Time
Password (OTP) using the Synthetics CLI; refer to <<elastic-synthetics-totp-command>>.
```

File: docs/en/serverless/synthetics/synthetics-mfa.mdx [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+Synthetics supports testing websites secured by Time-based One-Time Password
+(TOTP), a common MFA method that provides short lived One Time tokens that are
+desirable for enhancing security. 
+
+## Configuring using TOTP
+
+To test a browser journey that uses TOTP for MFA, you need to first configure the
+Synthetics authenticator token in your application. To do that, you can generate a One
+Time Password (OTP) using the Synthetics CLI, refer to <DocLink slug="/serverless/observability/synthetics-command-reference" section="@elastic/synthetics totp <secret>">.
```
Comment (bmorelli25) [inline docs/en/serverless/synthetics/synthetics-mfa.mdx:27]:
```suggestion
To test a browser journey that uses TOTP for MFA, first configure the
Synthetics authenticator token in the target application. To do this, generate a One-Time
Password (OTP) using the Synthetics CLI; refer to <DocLink slug="/serverless/observability/synthetics-command-reference" section="@elastic/synthetics totp <secret>">.
```

File: docs/en/observability/synthetics-mfa.asciidoc [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+journey('MFA Test', ({ page, params }) => {
+  step('Login using TOTP token', async () => {
+    // login using username and pass and go to 2FA in next page
+    const token = mfa.token(params.MFA_GH_SECRET);
+    await page.getByPlaceholder("token-input").fill(token)
+  });
+});
+```
+
+For monitors that are created from the Synthetics UI using the Script editor, `mfa` object can be accessed as shown below:
```
Comment (bmorelli25) [inline docs/en/observability/synthetics-mfa.asciidoc:48]:
```suggestion
For monitors created in the Synthetics UI using the Script editor, the `mfa` object can be accessed as shown below:
```

File: docs/en/serverless/synthetics/synthetics-mfa.mdx [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+journey("MFA Test", ({ page, params }) => {
+  step("Login using TOTP token", async () => {
+    // login using username and pass and go to 2FA in next page
+    const token = mfa.token(params.MFA_GH_SECRET);
+    await page.getByPlaceholder("token-input").fill(token);
+  });
+});
+```
+
+For monitors that are created from the Synthetics UI using the Script editor, `mfa` object can be accessed as shown below:
```
Comment (bmorelli25) [inline docs/en/serverless/synthetics/synthetics-mfa.mdx:53]:
```suggestion
For monitors created in the Synthetics UI using the Script editor, the `mfa` object can be accessed as shown below:
```

File: docs/en/observability/synthetics-mfa.asciidoc (hunk 5 above)
Comment (bmorelli25) [inline docs/en/observability/synthetics-mfa.asciidoc:17]: What about this instead?

```suggestion
== Configuring TOTP for MFA
```

File: docs/en/observability/synthetics-mfa.asciidoc (hunk 6 above)
Comment (bmorelli25) [inline docs/en/observability/synthetics-mfa.asciidoc:30]: What about this instead?

```suggestion
== Applying the TOTP Token in Browser Journeys
```

File: docs/en/serverless/synthetics/synthetics-mfa.mdx [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+Multi-factor Authentication (MFA) adds an essential layer of security to
+applications login processes, protecting against unauthorized access. A very
+common usecase in Synthetics is testing user journeys that involves websites
+protected behind MFA.
+
+Synthetics supports testing websites secured by Time-based One-Time Password
+(TOTP), a common MFA method that provides short lived One Time tokens that are
+desirable for enhancing security. 
+
+## Configuring using TOTP
```
Comment (bmorelli25) [inline docs/en/serverless/synthetics/synthetics-mfa.mdx:23]:
```suggestion
## Configuring TOTP for MFA
```

File: docs/en/serverless/synthetics/synthetics-mfa.mdx [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+Time Password (OTP) using the Synthetics CLI, refer to <DocLink slug="/serverless/observability/synthetics-command-reference" section="@elastic/synthetics totp <secret>">.
+
+```sh
+npx @elastic/synthetics totp <secret>
+
+// prints
+OTP Token: 123456
+```
+
+## Using TOTP token inside the journeys
```
Comment (bmorelli25) [inline docs/en/serverless/synthetics/synthetics-mfa.mdx:36]:
```suggestion
## Applying the TOTP Token in Browser Journeys
```

**Review by bmorelli25 — COMMENTED** [review]
Thanks for updating both content sets! I have a few recommendations.

**Comment by vigneshshanmugam** [comment]
> Currently each section is on its own page ([preview](https://observability-docs_bk_4444.docs-preview.app.elstc.co/guide/en/observability/master/synthetics-mfa.html)). I think they can go on one page (using the discrete tags below). What do you think?

That was the intention, Thanks for catching it.

**Comment by elasticdocs** [comment]
:rocket: Built _elastic-dot-co-docs-preview-docs_ successfully!

* https://elastic-dot-co-docs-production-2e4t1jdls-elastic-dev.vercel.app

* Built with commit 9f73e6ebc94453acde2fdd340407c06d99f83f27

> Issues? Visit #next-docs in Slack

**Pull request merged**